From bd274df592a11227b59e8effa6965a09e50cd3d2 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Mon, 9 Dec 2013 13:42:28 -0800
Subject: Fix EXTRA_KERNEL_MODULES build when more than one is defined

When the list is e.g. "a b" (or in this case "iwlwifi iwlwifi" in my
tree, which seems like a separate bug), this foreach expression
expands to "make a make b" and the build fails because it doesn't know
how to build "make" as a target.  There's no point in spawning a
separate make for those anyway that I can see.

Change-Id: Id1077435342bae470e95ea1576dd849215a0b3b3
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 core/kernel.mk |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/core/kernel.mk b/core/kernel.mk
index c771675..77a7f80 100644
--- a/core/kernel.mk
+++ b/core/kernel.mk
@@ -164,7 +164,7 @@ endef
 define make-modules
 	$(mk_kernel) INSTALL_MOD_PATH=$(1) modules_install
 	$(foreach item,$(dir $(EXTERNAL_KERNEL_MODULES_TO_INSTALL)),$(call make-ext-module,$(item),$(1)))
-	$(foreach item,$(EXTRA_KERNEL_MODULES), make $(item))
+	+ $(hide) $(MAKE) $(EXTRA_KERNEL_MODULES)
 	$(hide) rm -f $(1)/lib/modules/*/{build,source}
 	$(hide) cd $(1)/lib/modules && find -type f -print0 | xargs -t -0 -I{} mv {} .
 endef
-- 
1.7.4.1

